src/webgpu/api/operation/resource_init/depth_stencil_attachment_clear.spec.ts

```diff
@@ -0,0 +1,242 @@
+      subresourceRange
+    )) {
+      const width = this.textureWidth >> viewDescriptor.baseMipLevel!;
+      const height = this.textureHeight >> viewDescriptor.baseMipLevel!;
+
+      const renderTexture = this.device.createTexture({
+        size: [width, height, 1],
+        format: 'r8unorm',
+        usage: C.TextureUsage.OutputAttachment | C.TextureUsage.CopySrc,
+        sampleCount: this.params.sampleCount,
+      });
+
+      let resolveTexture = undefined;
+      let resolveTarget = undefined;
+      if (this.params.sampleCount > 1) {
+        resolveTexture = this.device.createTexture({
+          size: [width, height, 1],
+          format: 'r8unorm',
+          usage: C.TextureUsage.OutputAttachment | C.TextureUsage.CopySrc,
+        });
+        resolveTarget = resolveTexture.createView();
+      }
+
+      const commandEncoder = this.device.createCommandEncoder();
+      const pass = commandEncoder.beginRenderPass({
+        colorAttachments: [
+          {
+            attachment: renderTexture.createView(),
+            resolveTarget,
+            loadValue: [0, 0, 0, 0],
+            storeOp: 'store',
+          },
+        ],
+        depthStencilAttachment: {
+          attachment: texture.createView(viewDescriptor),
+          depthStoreOp: 'store',
+          depthLoadValue: 'load',
+          stencilStoreOp: 'store',
+          stencilLoadValue: 'load',
+        },
+      });
+
+      switch (this.params.readMethod) {
+        case ReadMethod.DepthTest:
+          pass.setPipeline(
+            this.getDepthTestReadbackPipeline(state, this.params.format, this.params.sampleCount)
+          );
+          break;
+
+        case ReadMethod.StencilTest:
+          pass.setPipeline(
+            this.getStencilTestReadbackPipeline(this.params.format, this.params.sampleCount)
+          );
+          pass.setStencilReference(initializedStateAsStencil(state));
+          break;
+
+        default:
+          unreachable();
+      }
+      pass.draw(3, 1, 0, 0);
+      pass.endPass();
+
+      const { bytesPerRow, byteLength, rowsPerImage } = getTextureCopyLayout('r8unorm', '2d', [
+        width,
+        height,
+        1,
+      ]);
+      const expectedTexelData = getTexelDataRepresentation('r8unorm').getBytes({ R: 1 });
+
+      const buffer = this.device.createBuffer({
+        size: byteLength,
+        usage: C.BufferUsage.CopySrc | C.BufferUsage.CopyDst,
+      });
+
+      commandEncoder.copyTextureToBuffer(
+        { texture: resolveTexture || renderTexture },
+        {
+          buffer,
+          bytesPerRow,
+          rowsPerImage,
+        },
+        [width, height, 1]
+      );
+
+      this.queue.submit([commandEncoder.finish()]);
+
+      const arrayBuffer = new ArrayBuffer(byteLength);
+      fillTextureDataWithTexelValue(expectedTexelData, 'r8unorm', '2d', arrayBuffer, [
+        width,
+        height,
+        1,
+      ]);
+      this.expectContents(buffer, new Uint8Array(arrayBuffer));
```

Comment:
A lot of this looks like maybe it could be a general expectSingleColorTextureContents helper. Could you consider whether that could be abstracted in a followup if it would be used elsewhere?

Reply:
We can probably do that. Probably on master first..